zppy/e3sm_diags.py

```diff
@@ -99,7 +103,7 @@ def e3sm_diags(config: ConfigObj, script_dir: str, existing_bundles, job_ids_fil
                         # Note that this line should still be executed even if jobid == -1
                         # The later tc_analysis-using e3sm_diags tasks still depend on this task (and thus will also fail).
                         # Add to the dependency list
-                        dependencies.append(status_file)
+                        carried_over_dependencies.append(status_file)
```

Comment:
Maybe rather than `carried_over_dependencies` we have a variable `previous_status_file` and then `if "tc_analysis" in c["sets"]` we add that to the dependencies as well. Then, diags not using tc_analysis wouldn't be affected.